### DISCUSSION TITLE
GTFS-RT Reference To Static GTFS Feeds

### DISCUSSION THREAD

**Discussion by AlexVPerl** [transitland/transitland#420] [Q&A] [answered]
Greetings, 

Some GTFS-RT feeds require their Static GTFS companion feeds in order to be processed. I'm wondering if Transitland shows or captures this relationship somewhere. 

Does the Feeds API provide GTFS-RT Reference to its parent Static GTFS feed it is bound to? If not, what logic or mechanism can be employed to find the same?

Thanks.

**Comment by drewda** [comment, marked as answer]
Yes, Transitland uses "operators" to group together related static GTFS and GTFS Realtime feeds. See more in the documentation: https://www.transit.land/documentation/concepts/agencies-and-operators/ and https://www.transit.land/documentation/atlas#operators

**Comment by AlexVPerl** [comment]
> Yes, Transitland uses "operators" to group together related static GTFS and GTFS Realtime feeds. See more in the documentation: https://www.transit.land/documentation/concepts/agencies-and-operators/ and https://www.transit.land/documentation/atlas#operators

Thank you for replying. Operator concept is great way to group multiple Agency IDs for a single operator. This works great if an operator has 1 static feed and 1 real-time feed.

There are also scenarios where some Operators have multiple feeds.

For example MTA New York City Transit (o-dr5r-nyct) has 18 feeds (8 static & 10 real-time): 
```
1. f-dr5r-nyctsubway, 
2. f-dr5r-mtanyctbusbrooklyn, 
3. f-dr5r-mtanyctbusmanhattan, 
4. f-dr5r-mtanyctbusstatenisland, 
5. f-dr5x-mtanyctbusqueens, 
6. f-dr72-mtanyctbusbronx, 
7. f-dr5r-mtabc, 
8. f-dr5r-mtanewyorkcitytransit, 
9. f-mta~nyc~rt~alerts, 
10. f-mta~nyc~rt~subway~1~2~3~4~5~6~7, 
11. f-mta~nyc~rt~subway~a~c~e, 
12. f-mta~nyc~rt~subway~b~d~f~m, 
13. f-mta~nyc~rt~subway~g, 
14. f-mta~nyc~rt~subway~j~z, 
15. f-mta~nyc~rt~subway~l, 
16. f-mta~nyc~rt~subway~n~q~r~w, 
17. f-mta~nyc~rt~subway~sir, 
18. f-mta~nyc~rt~bustime
```

I also noticed that some static & real-time feeds have almost identical OnestopId's for exception of the `"~rt"` suffix.  For example: `"f-sportran" and "f-sportran~rt"`

So I was looking into sub-stringing the OnestopId to match them but this doesn't work for all scenarios since OnestopIds can also be different as in the example of NYC MTA above.

Just wondering if there's a direct Static-to-Realtime feed mapping. Or if it can be inferred in some way, even perhaps a combination of matching methods. I know its not an easy problem to solve, so any feedback would be much appreciated. 

Thanks!

**Comment by michaelkirk** [comment]
For context, I originally posted transitland/transitland-atlas#2140, but was told to move the discussion here.

I think I'm trying to solve the same problem as here: to programmatically ask: "given a static feed (e.g. `f-9q5-metro~losangeles~rail`), give me its RT feeds (e.g. `f-metro~losangeles~alerts` and `f-metro~losangeles~rail~rt`)."

My takeaway from @drewda's response is that this is complicated. The best you can do is get a set of candidates via the operator record. As in my example and in this discussion, that's often not sufficient. 

The suggestion from @drewda was to further narrow down operator candidates by downloading each RT feed and matching to trip/stop IDs against the static feeds.

I can see how that could work. However, it seems like a good opportunity to centralize effort here in the atlas, by explicitly linking static feeds to their RT counterparts.

Some challenges:
 - It would need a schema change, as far as I can tell.
 - The relationship is many-to-many, not 1-to-1 (Some RT feeds cover multiple static feeds). Not necessarily a problem, but something to be aware of.
 - actually capturing the relationships could be a lot of work, though some code (as described above) might make the process smoother.

Is there any appetite for this?
